core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1170,6 +1180,25 @@ private static ByteString binaryOperator(
     return new ByteString(result);
   }
 
+  /**
+   * Utility for bitwise function applied to byteString and long values.
+   *
+   * @param b0 The first byteString value operand of bitwise function.
+   * @param b1 The second long value operand of bitwise function.
+   * @param bitOp BitWise binary operator.
+   * @return ByteString after bitwise operation.
+   */
+  private static ByteString binaryOperator(
+      ByteString b0, long b1, BinaryOperator<Byte> bitOp) {
+    final byte[] bytes0 = b0.getBytes();
+
+    final byte[] result = new byte[bytes0.length];
```

Comment:
Could you please remove `result` allocation and replace it with `bytes0`?

Reply:
Yes, Updated.